tests/api/reports-customers.php

```diff
@@ -550,9 +550,110 @@ public function test_sync_latest_order_customer() {
 		$response = $this->server->dispatch( $request );
 		$reports  = $response->get_data();
 
+		$customer_index = array_search( 'admin@example.org', array_column( $reports, 'email' ), true );
 		$this->assertEquals( 200, $response->get_status() );
-		$this->assertFalse( 'Random' === $reports[0]['city'] );
-		$this->assertFalse( 'FL' === $reports[0]['state'] );
-		$this->assertFalse( '54321' === $reports[0]['postcode'] );
+		$this->assertNotEquals( 'Random', $reports[ $customer_index ]['city'] );
+		$this->assertNotEquals( 'FL', $reports[ $customer_index ]['state'] );
+		$this->assertNotEquals( '54321', $reports[ $customer_index ]['postcode'] );
+	}
+
+	/**
+	 * Test sync order update with customer latest order info.
+	 */
+	public function test_sync_latest_order_customer_with_multiple_customers() {
+		wp_set_current_user( $this->user );
+
+		$order = WC_Helper_Order::create_order( 0 );
+		$order->set_status( 'completed' );
+		$order->set_total( 100 );
+		$order->save();
+		$order2 = WC_Helper_Order::create_order( 0 );
+		$order2->set_status( 'completed' );
+		$order2->set_total( 100 );
+		$order2->save();
+		$order3 = WC_Helper_Order::create_order( 0 );
+		$order3->set_status( 'completed' );
+		$order3->set_total( 100 );
+		$order3->set_billing_email( 'different@example.org' );
+		$order3->save();
+
+		WC_Helper_Queue::run_all_pending();
+
+		$customer_id  = CustomersDataStore::get_existing_customer_id_from_order( $order );
+		$customer2_id = CustomersDataStore::get_existing_customer_id_from_order( $order2 );
+		$customer3_id = CustomersDataStore::get_existing_customer_id_from_order( $order3 );
+		$this->assertEquals( $customer_id, $customer2_id );
+		$this->assertNotEquals( $customer_id, $customer3_id );
+		// update order info.
+		$order3->set_billing_city( 'Random' );
+		$order3->set_billing_state( 'FL' );
+		$order3->set_billing_postcode( '54321' );
+		$order3->save();
+
+		WC_Helper_Queue::run_all_pending();
+
+		$result = CustomersDataStore::sync_order_customer( $order3->get_id() );
+
+		WC_Helper_Queue::run_all_pending();
+
+		// Didn't update anything.
+		$this->assertEquals( -1, $result );
```

Comment:
Why test this? So this is correctly returning `-1` because Customer 3's address was actually updated in the call to `run_all_pending()` above?

Can we just omit lines 595-600?

Reply:
@jeffstieler good catch actually, the result should be `1` not `-1`, and it should of failed. It seems that assertEquals thinks both -1 and 1 are the same thing.... (I tried it with 1 and it still passed).

It is suppose to update correctly that's why I check the `$result`. The checks further down are to make sure that it indeed actually updated.

Reply:
Going to use `assertTrue( 1 === $result)` instead, that seems to compare correctly.

Reply:
Nvm, sorry for the many comments. I am still confused why the isEquals didn't work, but $result is suppose to return true. so the `1 ===` is not necessary.